ports/nrf/mphalport.c

```diff
@@ -35,6 +35,113 @@
 #include "nrfx_errors.h"
 #include "nrfx_config.h"
 
+#if MICROPY_PY_TIME_TICKS
+#include "nrfx_rtc.h"
+#include "nrf_clock.h"
+#endif
+
+
+#if MICROPY_PY_TIME_TICKS
+// use RTC1 for time ticks generation
+// with 32kHz tick resolution and overflow
+// handling in irq
+
+nrfx_rtc_t rtc1 = NRFX_RTC_INSTANCE(1);
+volatile mp_uint_t rtc_overflows = 0;
+
+const nrfx_rtc_config_t rtc_config_time_ticks = {
+    .prescaler = 0,
+    .reliable = 0,
+    .tick_latency = 0,
+    #ifdef NRF51
+    .interrupt_priority = 1,
+    #else
+    .interrupt_priority = 3,
+    #endif
+};
+
+STATIC void rtc_irq_time(nrfx_rtc_int_type_t event) {
+    // irq handler for overflow
+    if (event == NRFX_RTC_INT_OVERFLOW) {
+        rtc_overflows += 1;
+    }
+    // irq handler for wakeup from WFI (~1msec)
+    if (event == NRFX_RTC_INT_COMPARE0) {
+        nrfx_rtc_cc_set(&rtc1, 0, nrfx_rtc_counter_get(&rtc1) + 33, true);
+    }
+}
+
+void rtc1_init_time_ticks(void) {
+    // Start the low-frequency clock (if it hasn't been started already)
+    if (!nrf_clock_lf_is_running(NRF_CLOCK)) {
+        nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_LFCLKSTART);
+    }
+    nrfx_rtc_init(&rtc1, &rtc_config_time_ticks, rtc_irq_time);
+    nrfx_rtc_overflow_enable(&rtc1, true);
+    // generate a ~1msec irq to leave _WFI
+    nrfx_rtc_cc_set(&rtc1, 0, 30, true);
```

Comment:
the "30" should be the same as the "33" above, right?  in which case use the same #define constant

Reply:
that would be the first CC irq to schedule, so can actually be different? 

But I haven't looked at it closely. It seems that nrfx_rtc_init does not clear the counter state (as I have assumed). At least I didn't find any reference to it.

So I believe we should reset the counter using `nrfx_rtc_counter_clear`, as pointed below from @glennrub  of use a similar logic like `nrfx_rtc_counter_get(&rtc1) + xxx` ? My preference would be the second one.

Reply:
> My preference would be the second one.

Sounds good.